### PR TITLE
Fix vulnerabilities

### PR DESCRIPTION
**Pull request by DaniilShmelev**
**Task name**: root of the repo, ANTV1, AndroidSigningV2, AndroidSigningV3, ArchiveFilesV2, BashV3, CMakeV1, CUrlUploaderV2, CmdLineV2, CocoaPodsV0, DecryptFileV1, CopyFilesOverSSHV0

**Description**:
Root:
- Removed unused `mocha-junit-reporter`.
- Removed `mocha-junit-reporter` traces from `make.js`.
- Removed unused `markdown-toc`.
- Updated `mocha` to 5.x to resolve most vulnerabilities. There's still one remaining for this package - `mocha > mkdirp > minimist`. All dependencies are fixed only in mocha 9.x, but we can't update to this version until we fully migrate to node 10.
- Bumped other vulnerable packages. Some of them had breaking changes, so changes have been made to `make.js` and `make-util.js` to bring dependency usage up to date.

ANTV1, AndroidSigningV2, AndroidSigningV3, ArchiveFilesV2, BashV3, CMakeV1, CUrlUploaderV2, CmdLineV2, CocoaPodsV0, DecryptFileV1
- Bumped `path-parse`.
- For some of those tasks package-lock structure has changed, but no dependencies other than `path-parse` changed.

CopyFilesOverSSHV0
- Bumped vulnerable `ssh2` package version. This is a major version bump, so rigorous testing is needed.
- Had to bump `ssh2-sftp-client` too because of `ssh2` update.

**Documentation changes required:** No

**Added unit tests:** No

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by anatolybolshakov — APPROVED** [review]

**Comment by DaniilShmelev** [comment]
All tests successfully passed: https://dev.azure.com/v-dshmelev/playground/_build/results?buildId=1873

**Review by golubev-ivan — APPROVED** [review]
\

File: Tasks/CondaEnvironmentV0/Tests/L0_conda.ts [hunk 1]
```diff
@@ -122,13 +122,13 @@ it('fails if `conda` is not found', async function (done: Mocha.Done) {
...
+        throw new Error('should not have succeeded');
     } catch (e) {
         assert.strictEqual(e.message, 'loc_mock_CondaNotFound');
         assert(findConda.calledOnceWithExactly(Platform.Windows));
         assert(prependCondaToPath.notCalled);
         assert(createEnvironment.notCalled);
         assert(activateEnvironment.notCalled);
-        done();
     }
 });
```
Comment (golubev-ivan) [inline Tasks/CondaEnvironmentV0/Tests/L0_conda.ts:134]: Could you please add EOL?

Comment (DaniilShmelev) [reply Tasks/CondaEnvironmentV0/Tests/L0_conda.ts]: Fixed, thanks!

**Pull request merged**